sequencer/src/state_signature.rs

```diff
@@ -0,0 +1,27 @@
+use crate::{context::SequencerContext, network, Leaf, Node, SeqTypes};
+use hotshot_types::{light_client::LightClientState, traits::state::ConsensusTime};
+
+pub mod signature_pool;
+
+pub type StateSignatureScheme =
+    jf_primitives::signatures::schnorr::SchnorrSignatureScheme<ark_ed_on_bn254::EdwardsConfig>;
+pub use hotshot_stake_table::vec_based::config::FieldType as BaseField;
+
+pub async fn state_signature_hook<N: network::Type>(
+    context: &SequencerContext<SeqTypes, Node<N>>,
+    leaf: &Leaf,
+) {
+    let new_light_client_state = LightClientState::<BaseField> {
+        view_number: leaf.get_view_number().get_u64() as usize,
+        block_height: leaf.get_height() as usize,
+        block_comm_root: BaseField::default(),
```

Comment:
We should add a TODO here as well right?

Reply:
I just created 2 new issues. https://github.com/EspressoSystems/espresso-sequencer/issues/923, https://github.com/EspressoSystems/espresso-sequencer/issues/924